src/services/effects.ts

```diff
@@ -90,25 +90,27 @@ export function withEffects(store: Store<Actions>) {
     store.on('standard').startWith(store.get('standard'))
     )
     .subscribe(async ([providers, representativePoints, measure, standard]) => {
-      let adequacies = await getAdequacies(providers.map(_ => _.id), store.get('serviceAreas'))
-      store.set('adequacies')(
-        chain(representativePoints.map(_ => _.id))
-          .zipObject(adequacies)
-          .mapValues(_ => ({
-            isAdequate: isAdequate(
-              _.distance_to_closest_provider,
-              _.time_to_closest_provider,
-              measure,
-              standard
-            ),
-            id: _.id,
-            distanceToClosestProvider: _.distance_to_closest_provider,
-            timeToClosestProvider: _.time_to_closest_provider,
-            closestProviderByDistance: _.closest_provider_by_distance,
-            closestProviderByTime: _.closest_provider_by_time
-          }))
-          .value()
-      )
+      if (providers.length && representativePoints.length) {
```

Comment:
Does this mean that if:

1. User sets providers
2. User sets service areas
3. Adequacies are computed
4. User clears providers or service areas
5. Old adequacies are not cleared <-- Bug

Why is the `if` needed?

Reply:
I was getting errors on the backend because when providers was updated to nothing it was sending off a request to the API for adequacies. I was also getting an error because if there weren't any representative points and providers was updated it was sending a request for adequacies. Shouldn't cause a 500 error on the backend but it seems like we also shouldn't be computing this if we don't have providers or if we don't have representative points

Reply:
See also [issue 90 in the other repo](https://github.com/bayesimpact/time-distance-standards/issues/90).

Reply:
I'm actually getting that if I clear inputs I get the right message on the analytics drawer when I go back  
![image](https://user-images.githubusercontent.com/11825994/33468794-b0336ff2-d613-11e7-8415-041c51fa09d5.png) although it does seem like that could lead to a bug! Do you think it's better to clear adequacies if I'm clearing provider or service area? or do you think we should just be sending empty lists in our API requests?

Reply:
Update, the donut chart disappears, but the colors for adequate/inadequate service points stays, so the bug does exist!

Reply:
I would send the 2 empty arrays - we memoize API responses so the impact on performance/extra requests will be negligible.

Reply:
👍 Added TODO. Going to merge as is because Vivian has a demo this morning, then fix it!